app/assets/javascripts/test_executions.js

```diff
@@ -5,7 +5,46 @@ ready = function() {
     event.preventDefault();
     $("#new_test_execution").submit();
   });
+
+  /* enable each error popover */
+  $('.error-popup-btn').popover();
+
+  /* change 'view' and 'hide' text for popover buttons */
+  $('.error-popup-btn').on('show.bs.popover', function () {
+    this.children[0].innerText = this.children[0].innerText.replace('view', 'hide');
+  });
+  $('.error-popup-btn').on('hide.bs.popover', function () {
+    this.children[0].innerText = this.children[0].innerText.replace('hide', 'view');
+  });
+
+  /* link subnav with the execution_error_link s */
+  var navigation =  $("#xml_frame .subnav").navigator({targets: "a.execution_error_link", action: show_error_popup_and_jump});
+  $("#xml_frame").fixedHeader();
+
+  /* used to set index on navigation bar when user clicks a execution_error_link and not a navbar button */
+  $("#display_execution_results").on('click', 'a.execution_error_link', function(event) {
+    href = $(this).attr('href');
+    show_error_popup_and_jump(href);
+    navigation.data('navigator').setIndex(href);
+    return false;
+  });
+
+  function show_error_popup_and_jump(href) {
+    if ($(href)) {
+      $('button.error-popup-btn').popover('hide'); /* hide all error popovers before showing the new one */
+
+      var scroll_time = 300;      /* (milisec) time it takes to scroll from one error popover to another */
+      var highlight_time = 2000;  /* (milisec) time highlight lasts for individual error in error popover */
+      var pixels_down_page = 500; /* number of pixels down the page the error will apear after scrolling */
```

Comment:
where does this number come from? Will this work ok on every browser/resolution or would it be better to calculate this height based on the heights of certain elements or somesuch?